scipy/io/matlab/miobase.py

```diff
@@ -295,10 +295,10 @@ def matdims(arr, oned_as='column'):
     shape = arr.shape
     if shape == ():  # scalar
         return (1,1)
-    if functools.reduce(operator.mul, shape) == 0:  # zero elememts
-        return (0,) * np.max([arr.ndim, 2])
     if len(shape) == 1:  # 1D
-        if oned_as == 'column':
+        if functools.reduce(operator.mul, shape) == 0:  # zero elememts
```

Comment:
I realize this is an existing check, but it's weird and if `shape` has length 1 it's even more weird. Why not just `arr.size == 0` or `shape[0] == 0`?